contrib/uptrace/bun/bun_test.go

```diff
@@ -0,0 +1,246 @@
+	"testing"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/uptrace/bun"
+	"github.com/uptrace/bun/dialect/sqlitedialect"
+	_ "modernc.org/sqlite"
+)
+
+func TestMain(m *testing.M) {
+	_, ok := os.LookupEnv("INTEGRATION")
+	if !ok {
+		fmt.Println("--- SKIP: to enable integration test, set the INTEGRATION environment variable")
+		os.Exit(0)
+	}
+	os.Exit(m.Run())
+}
+
+func setupDB(opts ...Option) *bun.DB {
+	sqlite, err := sql.Open("sqlite", "file::memory:?cache=shared")
+	if err != nil {
+		panic(err)
+	}
+
+	db := bun.NewDB(sqlite, sqlitedialect.New())
+	Wrap(db, opts...)
+
+	return db
+}
+
+func TestImplementsHook(_ *testing.T) {
+	var _ bun.QueryHook = (*queryHook)(nil)
+}
+
+func TestSelect(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	db := setupDB()
+	parentSpan, ctx := tracer.StartSpanFromContext(context.Background(), "http.request",
+		tracer.ServiceName("fake-http-server"),
+		tracer.SpanType(ext.SpanTypeWeb),
+	)
+
+	var n, rows int64
+	// Using WithContext will make the postgres span a child of
+	// the span inside ctx (parentSpan)
+	res, err := db.NewSelect().ColumnExpr("1").Exec(ctx, &n)
+	parentSpan.Finish()
+	spans := mt.FinishedSpans()
+
+	require.NoError(t, err)
+	rows, _ = res.RowsAffected()
+	assert.Equal(int64(1), rows)
+	assert.Equal(2, len(spans))
+	assert.Equal(nil, err)
+	assert.Equal(int64(1), n)
+	assert.Equal("bun", spans[0].OperationName())
+	assert.Equal("http.request", spans[1].OperationName())
+	assert.Equal("uptrace/bun", spans[0].Tag(ext.Component))
+	assert.Equal(ext.DBSystemOtherSQL, spans[0].Tag(ext.DBSystem))
+}
+
+func TestServiceName(t *testing.T) {
+	t.Run("default", func(t *testing.T) {
+		assert := assert.New(t)
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		db := setupDB()
+		parentSpan, ctx := tracer.StartSpanFromContext(context.Background(), "http.request",
+			tracer.ServiceName("fake-http-server"),
+			tracer.SpanType(ext.SpanTypeWeb),
+		)
+
+		var n int
+		// Using WithContext will make the postgres span a child of
+		// the span inside ctx (parentSpan)
+		res, err := db.NewSelect().ColumnExpr("1").Exec(ctx, &n)
+		parentSpan.Finish()
+		spans := mt.FinishedSpans()
+
+		require.NoError(t, err)
+		rows, _ := res.RowsAffected()
+		assert.Equal(int64(1), rows)
+		assert.Equal(2, len(spans))
+		assert.Equal(nil, err)
+		assert.Equal(1, n)
+		assert.Equal("bun", spans[0].OperationName())
+		assert.Equal("http.request", spans[1].OperationName())
+		assert.Equal("bun.db", spans[0].Tag(ext.ServiceName))
+		assert.Equal("fake-http-server", spans[1].Tag(ext.ServiceName))
+		assert.Equal("uptrace/bun", spans[0].Tag(ext.Component))
+		assert.Equal(ext.DBSystemOtherSQL, spans[0].Tag(ext.DBSystem))
```

Comment:
```suggestion
		assert.Equal(ext.DBSystemOtherSQL, spans[0].Tag(ext.DBSystem))
		assert.Equal(spans[0].ParentID(), spans[1].SpanID())
```

(we should test the context is propagated correctly as well)